src/ui/classic/xcbinputwindow.cpp

```diff
@@ -47,10 +47,22 @@ void XCBInputWindow::updatePosition(InputContext *inputContext) {
     }
     int x, y, h;
 
-    x = inputContext->cursorRect().left();
-    y = inputContext->cursorRect().top();
+    // add support of input panel offset here.
+    // TODO: RTL support.
+    auto &theme = parent_->theme();
+    int leftSW, rightSW, topSW, bottomSW, expectedWidth, expectedHeight;
+    leftSW = theme.inputPanel->shadowMargin->marginLeft.value();
+    rightSW = theme.inputPanel->shadowMargin->marginRight.value();
+    topSW = theme.inputPanel->shadowMargin->marginTop.value();
+    bottomSW = theme.inputPanel->shadowMargin->marginBottom.value();
+
+    x = inputContext->cursorRect().left() - leftSW;
```

Comment:
I don't think you should change "x" and "y" here. Basically what we want to do is that

we have a rect with size (actualW*actualH) be put at (x,y).

The x,y calculation should not consider the shadow offset.

After all the existing calculation, we move the window by (-leftOffset, -topOffset).